torch/distributed/distributed_c10d.py

```diff
@@ -74,8 +77,14 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value, _ = getattr(Backend, name.upper())
         return value
 
+    @classmethod
+    def register_new_backend(cls, module_name, instantiating_interface_name):
```

Comment:
`register_backend` is good enough, `new` is implied.

Reply:
done